### ISSUE TITLE
Formbuilder group from value object with array

### ISSUE THREAD

**Issue by Fyro-Ing** [angular/angular#24182]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior
<pre><code>
    this.myObject = {myobject: [{id: 'test1'}, {id: 'test2'}]};
    this.myFormGroup = this.fb.group(this.myObject);
</code></pre>
RESULT
<pre><code>
    myFormGroup.value : { "myobject": { "id": "test1" } } 
</code></pre>

## Expected behavior
<pre><code>
myFormGroup.value : {myobject: [{id: 'test1'}, {id: 'test2'}]}
</code></pre>

## Minimal reproduction of the problem with instructions
https://stackblitz.com/edit/angular-fromgroup-with-array

## What is the motivation / use case for changing the behavior?
It seems to be a bug, FormGroup build FormControl tree without considering if it's an array or object

It'is ok with a FormArray

## Environment

<pre><code>
Angular version: 5.0.0
Angular version: 6.0.2

**Comment by kara** [comment]
Yeah, the problem here is that the API of `FormBuilder` allows you to build `FormControls` using an array of arguments. So if you pass an array, it assumes that you are trying to build a control with the value at index 0, the validator at index 1, etc.

For now, you can fix by wrapping your value as a `FormState`, e.g.  `myobject: {value: [...], disabled: false}`.

**Comment by Fyro-Ing** [comment]
Sample from Stackblitz updated with workaround

**Comment by martinsik** [comment]
@Fyro-Ing You could do also this which is even easier:
```
this.myFormGroup = this.fb.group({
  myobject: this.fb.control(this.myObject.myobject),
});
```

**Comment by martinsik** [comment]
I was trying to look into this but then realised that it probably can't assume if you're using an array as a value or you have an array without validators that might come from some preceding logic:

For example:
```
const loginField = ['some value'];

// some more logic that might call eg.: loginField.push(validator);

const g = b.group({
  'login': loginField,
  'password': ['some value', syncValidator, asyncValidator]
});
```

Now is `login` is supposed to be initialized with `['some value']` or just `'some value'`? Right now it'll be `'some value'` but if there was some additional logic in `FormBuilder.group()` that would consider it as a default value then the value of this form would be: `{login: ['some value'], password: "some value"}` or `{login: 'some value', password: "some value"}` which is even more confusing and unpredictable.

**Comment by Fyro-Ing** [comment]
In my use case, i use an optional initial object like 'user informations' to submit a form

-> with @Component for global user

`this.userForm = this.fb.group {
addresses: this.fb.array([]),
surname : [null]
 ect...
}`

-> @Component for addresses
` this.address.registerControl(address, ...);
this.address.registerControl(town, ...);
)}`
-> ect...

In user form, initial addresses values are push in addresses array form because patchValue "ignore" arrays

the goal is that each component manage (and initialize) specific controls, validators, ...

**Comment by epiphanydigital** [comment]
In an *ngFor this has the added symptom of binding multiple instances to the same model if the array value is identical.  So if two FormArray indexes both have a control by name "tags" (for example) and you pass in a value such as ['tag 1', 'tag 2', 'tag 3]... updating the binding from the view results in every instance of that value changing as well.  I believe this is a symptom of the same root cause.

My workaround for now, for anyone interested, was to use a simple getter/setter to stringify() prior to form model update and parse() before display to the view.

**Comment by adamdport** [comment]
I'm trying to initialize a control as an empty array, eg.

    let def= { foo: [] }
    let group = this.fb.group(def);

`group.value` is then `{ foo: null }` instead of the expected `{ foo: [] }`. I think this stems from the same issue?

**Comment by dylhunn** [comment]
As Kara suggests above, a workaround for now is to use the boxed `FormState` value (e.g. `{value: 'foo', disabled: false}`).

The fundamental issue is that `FormBuilder.group` is not typesafe, and accepts too many forms of argument. We should deprecate some of them (probably `[value, validator, validator]` and `controlConfig`). Unfortunately, that will involve design work and a schematic.

**Comment by dylhunn** [comment]
(An alternative to deprecating methods and removing them would be `FormBuilder2`, which I'm an advocate of.)

**Comment by Michael-Ziluck** [comment]
@Fyro-Ing Fairly certain that you just have a bug with the syntax here.

If you were to do the following instead, I'd suspect you'd have the intended result:

```typescript
this.myObject = { myObject: [ [{id: 'test1'}, {id: 'test2'}] ] };
this.myFormGroup = this.fb.group(this.myObject);
```

Note the addition of the extra set of `[]`.

**Comment by AlvaroP95** [comment]
Any news with this? It's been 6 years already...
